workers/ohsome_quality_analyst/indicators/ghs_pop_comparison_buildings/indicator.py

```diff
@@ -57,6 +58,9 @@ async def preprocess(self) -> bool:
         if query_results is None:
             return False
         self.feature_count = query_results["result"][0]["value"]
+        self.result.timestamp_osm = datetime.strptime(
+            query_results["result"][0]["timestamp"], "%Y-%m-%dT%H:%M:%S%z"
+        )
```

Comment:
Another solution could be. This is somewhat more readable because of the function name `fromisoformat`. Then I know that the timestamp is supposed to be in ISO format.

```suggestion
        timestamp = query_results["result"][0]["timestamp"].replace('Z', '+00:00')
        self.result.timestamp_osm = datetime.fromisoformat(timestamp).
```

Reference: https://discuss.python.org/t/parse-z-timezone-suffix-in-datetime/2220

Reply:
That's definitely possible, but [as the documentation says](https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat) I would recommend to not use the `.fromisoformat()` as it was only designed as inverse to `.isoformat()`. I would also not start to deal with time parsing manually (especially timezones). I would use [dateutil](https://dateutil.readthedocs.io/en/stable/parser.html#dateutil.parser.isoparse) instead.

Some examples:
```python
import dateutil.parser

timestamp = "2007-10-08T00:00:00Z
dateutil.parser.isoparse(timestamp)
# output: datetime.datetime(2007, 10, 8, 0, 0, tzinfo=tzutc())

timestamp = "2007-10-08T00:00:00+01:23"
dateutil.parser.isoparse(timestamp)
# output: datetime.datetime(2007, 10, 8, 0, 0, tzinfo=tzoffset(None, 4980))

Reply:
I tried to avoid additional dependencies. But since pandas and matplotlib are already depended on dateutil there no reason why we should not use it instead.